
# HG changeset patch
# User Sylvain Thénault <sylvain.thenault@logilab.fr>
# Date 1367930788 -7200
# Node ID 5a8fb4b4a7f567b65b8f935282316be0b1f9018e
# Parent  4e579e9a97922a94624c2d04ae1af5268f0d487c
[modutils] fix python3.3 crash on file_from_modpath. Closes #137244

diff -r 4e579e9a9792 -r 5a8fb4b4a7f5 ChangeLog
--- a/ChangeLog	Tue May 07 14:42:37 2013 +0200
+++ b/ChangeLog	Tue May 07 14:46:28 2013 +0200
@@ -2,8 +2,13 @@
 ============================
 
 --
-    * modutils: fix typo causing name error in python3 / bad message in python2
-      (#136037)
+    * modutils:
+
+      * fix typo causing name error in python3 / bad message in python2
+        (#136037)
+
+      * fix python3.3 crash in file_from_modpath due to implementation
+        change of imp.find_module wrt builtin modules (#137244)
 
 
 2013-04-16  --  0.59.1
diff -r 4e579e9a9792 -r 5a8fb4b4a7f5 modutils.py
--- a/modutils.py	Tue May 07 14:42:37 2013 +0200
+++ b/modutils.py	Tue May 07 14:46:28 2013 +0200
@@ -606,6 +606,15 @@
         checkeggs = False
     imported = []
     while modpath:
+        # take care to changes in find_module implementation wrt builtin modules
+        #
+        # Python 2.6.6 (r266:84292, Sep 11 2012, 08:34:23)
+        # >>> imp.find_module('posix')
+        # (None, 'posix', ('', '', 6))
+        #
+        # Python 3.3.1 (default, Apr 26 2013, 12:08:46)
+        # >>> imp.find_module('posix')
+        # (None, None, ('', '', 6))
         try:
             _, mp_filename, mp_desc = find_module(modpath[0], path)
         except ImportError:
@@ -613,7 +622,7 @@
                 return _search_zip(modpath, pic)[:2]
             raise
         else:
-            if checkeggs:
+            if checkeggs and mp_filename:
                 fullabspath = [abspath(x) for x in _path]
                 try:
                     pathindex = fullabspath.index(dirname(abspath(mp_filename)))

